src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -22,10 +23,30 @@
 )
 
 
+def _deprecated_memory_arg(shared_memory: bool, share_inputs: bool) -> bool:
+    if shared_memory is not None:
+        warnings.warn(
+            "`shared_memory` is deprecated and will be removed in 2024.0. "
+            "Value of `shared_memory` is going to override `share_inputs` value. "
```

Comment:
If person is using share_inputs then he is probably aware that this exists and also aware that it replaced shared_memory, so this information about overriding probably does not add much value here. Instead, if user uses this old flag - probably he is not aware that there is new one.
Could we just say - we recommend to use share_inputs instead?

Reply:
> this information about overriding probably does not add much value here

This information is only shown when `shared_memory` is explicitly used. Which actually make sense as only users of the old flag will be notified. (+ there is `"Please use only share_inputs explicitly."` at the end of message)

If the new flag `share_inputs` or positional argument is used, there is nothing to notify about. Function is ran without warning.

Overall the whole problem boils down to the priority of flags which do the same thing. In summary:

- (until deprecation) `shared_memory` will trigger a warning about overriding.
- anything like `infer(..., shared_memory=True, share_inputs=False)` wouldn't be consider undefined behavior.
- both `infer(..., True)` and `infer(..., share_inputs=True)` would run without warnings.